v2/internal/middleware/middleware.go

```diff
@@ -31,15 +33,15 @@ func (r *stack[IN, OUT]) AddMiddleware(mw Middleware[IN, OUT]) {
 	r.middlewares = append(r.middlewares, mw)
 }
 
-func (r *stack[IN, OUT]) Process(options IN) OUT {
-	var next func(item IN) OUT = nil
+func (r *stack[IN, OUT]) Process(ctx context.Context, options IN) OUT {
+	var next func(c context.Context, item IN) OUT = nil
 	middlewares := append(r.middlewares)
```

Comment:
```suggestion
	middlewares := r.middlewares
```

Reply:
This was actually a bug in the code, I meant to make a duplicate, not a reference copy, I'm fixing using copy method.

Reply:
This was fine. Assigning `r.middlewares` to `middlewares` makes a shallow copy of the slice, hence the index changes like `middlewares = middlewares[1:]` doesn't affect the `r.middlewares`.